src/org/thoughtcrime/securesms/components/ScaleStableImageView.java

```diff
@@ -0,0 +1,181 @@
+import android.util.AttributeSet;
+import android.util.Log;
+
+import com.bumptech.glide.load.engine.DiskCacheStrategy;
+
+import org.thoughtcrime.securesms.mms.GlideApp;
+import org.thoughtcrime.securesms.util.Util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.AppCompatImageView;
+
+import static android.content.res.Configuration.ORIENTATION_PORTRAIT;
+import static android.content.res.Configuration.ORIENTATION_LANDSCAPE;
+
+/**
+ * An image view that maintains the size of the drawable provided.
+ * That means: when you set an image drawable it will be scaled to fit the screen once.
+ * If you rotate the screen it will be rescaled to fit.
+ * If you crop the screen (e.g. because the soft keyboard is displayed) the image is cropped instead.
+ *
+ * @author Angelo Fuchs
+ */
+public class ScaleStableImageView
+    extends AppCompatImageView
+    implements KeyboardAwareLinearLayout.OnKeyboardShownListener, KeyboardAwareLinearLayout.OnKeyboardHiddenListener {
+
+    private static final String TAG = ScaleStableImageView.class.getSimpleName();
+
+    private Drawable defaultDrawable;
+    private Drawable currentDrawable;
+    private Map<String, Drawable> storedSizes = new HashMap<>();
+
+    public ScaleStableImageView(Context context) {
+        this(context, null);
+    }
+
+    public ScaleStableImageView(Context context, @Nullable AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public ScaleStableImageView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+    @Override
+    public void setImageDrawable(@Nullable Drawable drawable) {
+        defaultDrawable = drawable;
+        storedSizes.clear();
+        overrideDrawable(defaultDrawable);
+    }
+
+    private void overrideDrawable(Drawable newDrawable) {
+        if (currentDrawable == newDrawable) return;
+        currentDrawable = newDrawable;
+        super.setImageDrawable(newDrawable);
+    }
+
+    private int landscapeWidth = 0;
+    private int landscapeHeight = 0;
+    private int portraitWidth = 0;
+    private int portraitHeight = 0;
+    private boolean keyboardShown = false;
+
+    @Override
+    protected void onSizeChanged(int width, int height, int oldWidth, int oldHeight) {
+        if (width == 0 || height == 0) return;
+        final String newKey = width + "x" + height;
+        int orientation = getResources().getConfiguration().orientation;
+        boolean portrait;
+        if (orientation == ORIENTATION_PORTRAIT) {
+            portrait = true;
+        } else if (orientation == ORIENTATION_LANDSCAPE) {
+            portrait = false;
+        } else {
+            Log.i(TAG, "orientation was: " + orientation);
+            return; // something fishy happened.
+        }
+
+        measureViewSize(width, height, oldWidth, oldHeight, portrait);
```

Comment:
This line can be moved below the early return block (`if (!(defaultDrawable instanceof BitmapDrawable)) {`...)

Reply:
Agreed, I changed the code.